### PR TITLE
Add Tests for GSCodingFlags

### PR DESCRIPTION
**Pull request by johnathan-becker**
Due to recent changes around GSCodingFlags.h I have added a test that makes sure it works in the future

### CODE REVIEW

File: Tests/gui/GSCodingFlags/GSCellFlags.m [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    pass(mask.value == 0b00010010000000000001110000001001, "mask.flags translates to mask.value");
+#else
+    pass(mask.value == 0b10010000001110000000000001001000, "mask.flags translates to mask.value");
+#endif
+    // reset mask
+    mask.value = 0;
+    mask.flags = (GSCellFlags){0};
+    // now make sure values translate to flags
+#if GS_WORDS_BIGENDIAN == 1
+    mask.value = 0b00000001001000000000110100000001;
```
Comment (fredkiefer) [inline Tests/gui/GSCodingFlags/GSCellFlags.m:45]: Both values look wrong. I would expect that they are different and identical to the ones above

File: Tests/gui/GSCodingFlags/GSCellFlags.m [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include "GSCodingFlags.h"
+
+int main()
+{
+    CREATE_AUTORELEASE_POOL(arp);
+    GSCellFlagsUnion mask = { { 0 } };
+
+    START_SET("GSCodingFlags GNUstep CellFlags Union")
+    
+    NS_DURING
```
Comment (fredkiefer) [inline Tests/gui/GSCodingFlags/GSCellFlags.m:16]: I don't think that we need this check for this test. Nothing that would require a backend is used here.

File: Tests/gui/GSCodingFlags/GSCellFlags.m [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    mask.value = 0;
+    mask.flags = (GSCellFlags){0};
+    // now make sure values translate to flags
+#if GS_WORDS_BIGENDIAN == 1
+    mask.value = 0b00000001001000000000110100000001;
+#else
+    mask.value = 0b00000001001000000000110100000001;
+#endif
+
+    pass(mask.flags.state = 1, "state is correctly set");
```
Comment (fredkiefer) [inline Tests/gui/GSCodingFlags/GSCellFlags.m:39]: You are assigning the value 1 here not testing it. The same is true for the other checks. This explains why you have seven checks but only six bits that are set to one.

Comment (johnathan-becker) [reply Tests/gui/GSCodingFlags/GSCellFlags.m:39]: I see that aplogies. I have been out of a town for a few days but I did just update the test and I ran it locally and it passed, although I do not have access to a big endian system.

Comment (fredkiefer) [reply Tests/gui/GSCodingFlags/GSCellFlags.m:39]: Now the values are plausible, but the checks are still wrong. Could you please fix that? '=' => '=='

**Review by fredkiefer — APPROVED** [review]
Now it looks good

**Pull request merged**